### ISSUE TITLE
Packages automatically detected or specified with make(..., packages = ...) must actually be installed.

### ISSUE THREAD

**Issue by wlandau-lilly** [wlandau-lilly/drake#29]
Otherwise, `make()` could quit in error. This causes problems if you call `devtools::load_all("your_package")`  before `make()` because then `drake` detects and tries to load `"your_package"`. In the next patch I am preparing (unavoidably closed-source), I already fixed this problem by using `require()` rather than `library()` in `add_packages_to_prework()` so that a warning is generated rather than an error. But for now, I do not think this little hiccup is worth rushing another public release. If you want to call `devtools::load_all("your_package")`, put that line of code in the `prework` argument and manually set `packages` so that it excludes `"your_package"`.

```r
devtools::load_all("your_package") # Try to avoid load_all() in your regular setup code...
packages = "MASS" # ...otherwise, declare all your packages so drake does not search for them.
prework = 'devtools::load_all("your_package")' # It is best to use the prework for load_all().
make(my_plan, packages = packages, prework = prework)
```

**Comment by wlandau-lilly** [comment]
Maybe in the future drake should try `require()` first and then `devtools::load_all(pkg = path_to_package)` if that fails. But then I need to know `path_to_package`, so it may not be such a trivial fix.

**Comment by wlandau-lilly** [comment]
See #32. If #29 were already fixed, the error reported by @vlulla in #32 would have only been a warning.

**Comment by wlandau-lilly** [comment]
Fixed in v2.1.2.
